Google.Api.Generator.Tests/Invoker.cs

```diff
@@ -128,5 +134,35 @@ public static WithPath TempDir()
             Directory.CreateDirectory(dirPath);
             return new WithPath(() => Directory.Delete(dirPath, recursive: true), dirPath);
         }
+
+        private static Dictionary<string, string> GetOSSpecificConfigs(string rootDir)
```

Comment:
One refactoring to consider - if we're really only expecting the three values, consider returning a tuple instead:

```csharp
private static (string protoc, string runtime, string plugin) GetProtobufConfig() =>
    RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? ("protoc", "linux-x64", "Google.Api.Generator")
    : RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? ("protoc_macosx_x64", "osx-x64", "Google.Api.Generator" )
    : ("protoc.exe", "win-x64", "Google.Api.Generator.exe" );
```

Then I'd construct the full paths in common code in the static constructor.

Reply:
At that point, it might not even be worth having it as a separate method...